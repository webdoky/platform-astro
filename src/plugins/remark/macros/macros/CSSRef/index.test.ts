import type { Root } from 'mdast';
import { afterAll, beforeEach, describe, expect, it, vi } from 'vitest';

import type { AstroFile } from '../../../validate-astro-file.ts';
import type { AbstractMacroParentNode } from '../../types.ts';

import CSReference from './index.ts';

vi.mock('../../../../registry/get-all.ts', () => ({
  default: vi.fn().mockReturnValue([]),
}));
vi.mock('../../../../registry/has-page.ts', () => ({
  default: vi.fn().mockReturnValue(true),
}));

describe('CSSRef', () => {
  const OLD_ENV = process.env;

  beforeEach(() => {
    vi.resetModules(); // Most important - it clears the cache
    process.env = {
      TARGET_LOCALE: 'uk',
    };
  });

  afterAll(() => {
    process.env = OLD_ENV; // Restore old environment
  });
  it('should throw an error if the sidebar already exists', () => {
    expect(() => {
      const tree = {
        type: 'root',
        children: [
          {
            type: 'macro',
            name: 'CSSRef',
            parameters: [],
          },
        ],
      };
      const file = {
        data: {
          astro: {
            frontmatter: {
              sidebar: [],
              'page-type': '',
              title: '',
              slug: '',
            },
          },
        },
      } as any;
      CSReference(
        {
          type: 'macro',
          name: 'CSSRef',
          parameters: [],
        },
        0,
        tree,
        tree as Root,
        file,
      );
    }).toThrowError('Sidebar already exists');
  });
  it('should add a sidebar to the frontmatter', () => {
    const tree: AbstractMacroParentNode = {
      type: 'root',
      children: [
        {
          type: 'macro',
          name: 'CSSRef',
          parameters: [],
        },
      ],
    };
    const file = {
      data: {
        astro: {
          frontmatter: {},
        },
      },
    } as AstroFile;
    CSReference(
      {
        type: 'macro',
        name: 'CSSRef',
        parameters: [],
      },
      0,
      tree,
      tree as Root,
      file,
    );
    expect(file.data.astro.frontmatter.sidebar).toMatchSnapshot();
  });
});
