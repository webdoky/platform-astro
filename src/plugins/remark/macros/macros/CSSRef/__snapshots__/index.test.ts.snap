// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CSSRef > should add a sidebar to the frontmatter 1`] = `
[
  {
    "links": [],
    "sections": [],
  },
  {
    "links": [],
    "sections": [
      {
        "expanded": false,
        "items": [
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_animations/Using_CSS_animations",
            "title": "Застосування анімацій CSS",
          },
        ],
        "title": "Анімації",
      },
      {
        "expanded": false,
        "items": [
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_backgrounds_and_borders/Using_multiple_backgrounds",
            "title": "Застосування кількох фонів",
          },
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_backgrounds_and_borders/Resizing_background_images",
            "title": "Зміна розмірів фонових зображень",
          },
        ],
        "title": "Фони та рамки",
      },
      {
        "expanded": false,
        "items": [
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_box_alignment/Box_Alignment_In_Block_Abspos_Tables",
            "title": "Шикування блоку при блоковому компонуванні",
          },
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_box_alignment/Box_Alignment_in_Flexbox",
            "title": "Шикування блоку при флексбоксі",
          },
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_box_alignment/Box_Alignment_In_Grid_Layout",
            "title": "Шикування блоку при сітковому компонуванні",
          },
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_box_alignment/Box_Alignment_in_Multi-column_Layout",
            "title": "Шикування блоку при багатоколонковому компонуванні",
          },
        ],
        "title": "Шикування рамки",
      },
      {
        "expanded": false,
        "items": [
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_box_model/Introduction_to_the_CSS_box_model",
            "title": "Вступ до базової рамкової моделі CSS",
          },
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_box_model/Mastering_margin_collapsing",
            "title": "Опанування перекриття зовнішніх полів",
          },
        ],
        "title": "Рамкова модель",
      },
      {
        "expanded": false,
        "items": [
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_colors/Applying_color",
            "title": "Застосування кольору до елементів HTML за допомогою CSS",
          },
        ],
        "title": "Кольори",
      },
      {
        "expanded": false,
        "items": [
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_multicol_layout/Basic_concepts",
            "title": "Базові концепції багатоколонки",
          },
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_Multicol_Layout/Styling_columns",
            "title": "Оформлення колонок",
          },
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_Multicol_Layout/Spanning_balancing_columns",
            "title": "Заповнення та балансування",
          },
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_Multicol_Layout/Handling_overflow_in_multicol_layout",
            "title": "Обробка переповнення у багатоколонці",
          },
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_Multicol_Layout/Handling_content_breaks_in_multicol_layout",
            "title": "Розриви вмісту в багатоколонці",
          },
        ],
        "title": "Колонки",
      },
      {
        "expanded": false,
        "items": [
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_Conditional_Rules/Using_Feature_Queries",
            "title": "Застосування запитів можливостей",
          },
        ],
        "title": "Умовні директиви",
      },
      {
        "expanded": false,
        "items": [
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSSOM_View/Coordinate_systems",
            "title": "Координатні системи",
          },
        ],
        "title": "Представлення CSSOM",
      },
      {
        "expanded": false,
        "items": [
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_flexible_box_layout/Basic_concepts_of_flexbox",
            "title": "Основні концепції Флексбоксу",
          },
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_flexible_box_layout/Relationship_of_flexbox_to_other_layout_methods",
            "title": "Порівняння з іншими способами компонування",
          },
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_flexible_box_layout/Aligning_items_in_a_flex_container",
            "title": "Шикування елементів у гнучкому контейнері",
          },
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_flexible_box_layout/Ordering_flex_items",
            "title": "Упорядкування гнучких елементів",
          },
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_flexible_box_layout/Controlling_ratios_of_flex_items_along_the_main_axis",
            "title": "Контроль пропорцій гнучких елементів",
          },
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_flexible_box_layout/Mastering_wrapping_of_flex_items",
            "title": "Опанування перенесення гнучких елементів",
          },
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_flexible_box_layout/Typical_use_cases_of_flexbox",
            "title": "Типові випадки для застосування Флексбоксу",
          },
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_flexible_box_layout/Backwards_compatibility_of_flexbox",
            "title": "Зворотна сумісність Флексбоксу",
          },
        ],
        "title": "Флексбокс",
      },
      {
        "expanded": false,
        "items": [
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_flow_layout/Block_and_inline_layout_in_normal_flow",
            "title": "Блокове та рядкове компонування при звичайному плині",
          },
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_flow_layout/In_flow_and_out_of_flow",
            "title": "У плині та поза плином",
          },
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_flow_layout/Introduction_to_formatting_contexts",
            "title": "Пояснення контекстів форматування",
          },
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_flow_layout/Flow_layout_and_writing_modes",
            "title": "Потокове компонування та напрямки письма",
          },
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_flow_layout/Flow_layout_and_overflow",
            "title": "Потокове компонування та переповнення",
          },
        ],
        "title": "Потокове компонування",
      },
      {
        "expanded": false,
        "items": [
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_fonts/OpenType_fonts_guide",
            "title": "Посібник з можливостей шрифтів OpenType",
          },
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_fonts/Variable_fonts_guide",
            "title": "Посібник зі змінних шрифтів",
          },
        ],
        "title": "Шрифти",
      },
      {
        "expanded": false,
        "items": [
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_grid_layout/Basic_concepts_of_grid_layout",
            "title": "Основні ідеї розкладки за сіткою",
          },
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_grid_layout/Relationship_of_grid_layout_with_other_layout_methods",
            "title": "Зв'язок з іншими способами компонування",
          },
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_grid_layout/Grid_layout_using_line-based_placement",
            "title": "Розміщення на основі рядів",
          },
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_grid_layout/Grid_template_areas",
            "title": "Області шаблону сітки",
          },
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_grid_layout/Grid_layout_using_named_grid_lines",
            "title": "Компонування за допомогою іменованих рядів сітки",
          },
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_grid_layout/Auto-placement_in_grid_layout",
            "title": "Авторозміщення при сітковому компонуванні",
          },
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_grid_layout/Box_alignment_in_grid_layout",
            "title": "Шикування блоку при сітковому компонуванні",
          },
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_grid_layout/Grids_logical_values_and_writing_modes",
            "title": "Сітки, логічні значення та напрям письма",
          },
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_grid_layout/Grid_layout_and_accessibility",
            "title": "Сіткове компонування та доступність",
          },
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_grid_layout/Grid_layout_and_progressive_enhancement",
            "title": "Сіткове компонування та прогресивне покращення",
          },
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_grid_layout/Realizing_common_layouts_using_grids",
            "title": "Реалізація поширених макетів за допомогою сіток",
          },
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_grid_layout/Subgrid",
            "title": "Підсітка",
          },
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_grid_layout/Masonry_layout",
            "title": "Кладкове компонування",
          },
        ],
        "title": "Сітки",
      },
      {
        "expanded": false,
        "items": [
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_images/Using_CSS_gradients",
            "title": "Застосування градієнтів CSS",
          },
        ],
        "title": "Зображення",
      },
      {
        "expanded": false,
        "items": [
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_counter_styles/Using_CSS_counters",
            "title": "Застосування лічильників CSS",
          },
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_lists/Consistent_list_indentation",
            "title": "Узгоджені відступи у списках",
          },
        ],
        "title": "Списки та лічильники",
      },
      {
        "expanded": false,
        "items": [
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_logical_properties_and_values/Basic_concepts_of_logical_properties_and_values",
            "title": "Базові концепції",
          },
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_logical_properties_and_values/Floating_and_positioning",
            "title": "Пливуни та позиціювання",
          },
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_logical_properties_and_values/Margins_borders_padding",
            "title": "Зовнішні відступи, межі та внутрішні відступи",
          },
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_logical_properties_and_values/Sizing",
            "title": "Задання розмірів",
          },
        ],
        "title": "Логічні властивості",
      },
      {
        "expanded": false,
        "items": [
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_media_queries/Using_media_queries",
            "title": "Використання медіазапитів",
          },
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_media_queries/Using_media_queries_for_accessibility",
            "title": "Використання медіазапитів для потреб доступності",
          },
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_media_queries/Testing_media_queries",
            "title": "Програмне тестування медіазапитів",
          },
        ],
        "title": "Медіазапити",
      },
      {
        "expanded": false,
        "items": [
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_Nesting/Using_CSS_nesting",
            "title": "Застосування вкладеності в CSS",
          },
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_Nesting/Nesting_at-rules",
            "title": "Директиви вкладеності",
          },
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_Nesting/Nesting_and_specificity",
            "title": "Вкладеність і специфічність",
          },
        ],
        "title": "Вкладеність",
      },
      {
        "expanded": false,
        "items": [
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_positioned_layout/Understanding_z-index",
            "title": "Розуміння z-індексу у CSS",
          },
        ],
        "title": "Позиціювання",
      },
      {
        "expanded": false,
        "items": [
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_scroll_snap/Basic_concepts",
            "title": "Основні принципи захоплення прокрутки",
          },
        ],
        "title": "Захоплення прокручення",
      },
      {
        "expanded": false,
        "items": [
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_shapes/Overview_of_shapes",
            "title": "Фігури — огляд",
          },
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_shapes/From_box_values",
            "title": "Створення фігур на базі параметрів рамки",
          },
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_shapes/Basic_shapes",
            "title": "Основні фігури",
          },
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_shapes/Shapes_from_images",
            "title": "Створення фігур із зображень",
          },
        ],
        "title": "Фігури",
      },
      {
        "expanded": false,
        "items": [
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_text/Wrapping_breaking_text",
            "title": "Переноси та розриви тексту",
          },
        ],
        "title": "Текст",
      },
      {
        "expanded": false,
        "items": [
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_transforms/Using_CSS_transforms",
            "title": "Застосування перетворень",
          },
        ],
        "title": "Перетворення",
      },
      {
        "expanded": false,
        "items": [
          {
            "hasTranslation": true,
            "isCurrent": false,
            "path": "/uk/docs/Web/CSS/CSS_transitions/Using_CSS_transitions",
            "title": "Застосування переходів",
          },
        ],
        "title": "Переходи",
      },
    ],
  },
]
`;
